metrics/mean_iou/mean_iou.py

```diff
@@ -0,0 +1,237 @@
+        )
+        total_area_intersect += area_intersect
+        total_area_union += area_union
+        total_area_pred_label += area_pred_label
+        total_area_label += area_label
+    return total_area_intersect, total_area_union, total_area_pred_label, total_area_label
+
+
+def mean_iou(
+    results, gt_seg_maps, num_classes, ignore_index, nan_to_num=None, label_map=dict(), reduce_zero_label=False
+):
+    """Calculate Mean Intersection and Union (mIoU)
+
+    Args:
+        results (list[ndarray]):
+            List of prediction segmentation maps.
+        gt_seg_maps (list[ndarray]):
+            List of ground truth segmentation maps.
+        num_classes (int):
+            Number of categories.
+        ignore_index (int):
+            Index that will be ignored in evaluation.
+        nan_to_num (int, optional):
+            If specified, NaN values will be replaced by the numbers defined by the user. Default: None.
+        label_map (dict):
+            Mapping old labels to new labels. Default: dict().
+        reduce_zero_label (bool):
+            Whether to ignore the zero label. Default: False.
+
+     Returns:
+        dict[str, float | ndarray]:
+            <aAcc> float: Overall accuracy on all images.
+            <Acc> ndarray: Per category accuracy, shape (num_classes, ).
+            <IoU> ndarray: Per category IoU, shape (num_classes, ).
+    """
+    iou_result = eval_metrics(
+        results=results,
+        gt_seg_maps=gt_seg_maps,
+        num_classes=num_classes,
+        ignore_index=ignore_index,
+        metrics=["mIoU"],
+        nan_to_num=nan_to_num,
+        label_map=label_map,
+        reduce_zero_label=reduce_zero_label,
+    )
+    return iou_result
+
+
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Mean_IoU(datasets.Metric):
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "predictions": datasets.Sequence(datasets.Value("int32")),
+                    "gt_seg_maps": datasets.Sequence(datasets.Value("int32")),
+                    "num_classes": datasets.Value("int32"),
+                    "ignore_index:": datasets.Value("int32"),
+                    "nan_to_num": datasets.Value("int32"),
+                    "label_map": dict,
+                    "reduce_zero_label": datasets.Value("bool"),
+                }
+            ),
+            reference_urls=[
+                "https://github.com/open-mmlab/mmsegmentation/blob/71c201b1813267d78764f306a297ca717827c4bf/mmseg/core/evaluation/metrics.py"
+            ],
+        )
+
+    def _compute(
+        self,
+        predictions,
+        gt_seg_maps,
+        num_classes,
+        ignore_index,
+        nan_to_num=None,
+        label_map=dict(),
+        reduce_zero_label=False,
```

Comment:
Let's add some typing to the signature (and avoid mutable types):
```suggestion
        self,
        predictions,
        references,
        num_classes: int,
        ignore_index: bool,
        nan_to_num: Optional[int] = None,
        label_map: Optional[Dict[int, int]] = None,  # in the body: label_map = label_map if label_map is not None else {}
        reduce_zero_label: bool = False,
```